engines/query-sparql/package.json

```diff
@@ -168,6 +168,11 @@
     "@comunica/runner": "^2.4.0",
     "@comunica/runner-cli": "^2.4.0"
   },
+  "optionalDependencies": {
+    "@inrupt/solid-client-authn-core": "^1.12.2",
```

Comment:
I guess these should be devDependencies? (if so, they can be added as devdep in the root package.json)

Reply:
The reason, I made them optional was that it was failing the `yarn install --pure-lockfile` in the CI otherwise.
Should I add them to the `optionalDependencies` of the root `package.json` though?

Reply:
> The reason, I made them optional was that it was failing the yarn install --pure-lockfile in the CI otherwise.

Why was it failing exactly?

Reply:
Sorry, that might've been a bit unclear. `solid-client-authn-core` and `solid-client-authn-node` do not support node 18 yet. That's why the `QuerySparql-solid-test.ts` needed to be excluded from testing as well when in a node 18 environment, as the test depends on `solid-client-authn-*`.

Reply:
Ah, I see.

Wondering if there's a way (perhaps yarn-specific) to only install a dep if on a certain node version.

If not, then we have to resort to installing those deps in the CI.

The problem with optionalDeps is that end-users will also end up having these installed, while they are not needed.

Reply:
I found the following plugin: https://www.npmjs.com/package/optional-dev-dependency As I'm absolutely no expert with npm package management - do you think that's too hacky?

How would it work installing the deps in the CI? Adding the dependency with `yarn add` before running `yarn install`, removing it when on node 18, or something else?

Reply:
> I found the following plugin: https://www.npmjs.com/package/optional-dev-dependency As I'm absolutely no expert with npm package management - do you think that's too hacky?

Not sure, doesn't look very well maintained.

> How would it work installing the deps in the CI? Adding the dependency with yarn add before running yarn install, removing it when on node 18, or something else?

Yes, adding them after install on Node 18 could work!